### PR TITLE
JAVA-24275 Upgrade Keycloak for oauth-resource-server/authorization-server

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

File: oauth-resource-server/authorization-server/pom.xml [hunk 1]
```diff
@@ -90,6 +126,13 @@
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
 				<configuration>
+					<mainClass>com.baeldung.auth.AuthorizationServerApp</mainClass>
+					<requiresUnpack>
+						<dependency>
+							<groupId>org.keyclaok</groupId>
```
Comment (anastasiosioannidis) [inline oauth-resource-server/authorization-server/pom.xml:132]: Could you check the miss-spelling here? "keyclaok"

File: oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/SimplePlatformProvider.java
Comment (anastasiosioannidis) [inline oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/SimplePlatformProvider.java:1]: Could you briefly explain the changes here?

File: oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/Resteasy3Provider.java
Comment (anastasiosioannidis) [inline oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/Resteasy3Provider.java:1]: Could you briefly explain the changes here?

File: oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/SimplePlatformProvider.java
Comment (anuragkumawat) [reply oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/SimplePlatformProvider.java:1]: PlatformProvider interface in new keycloak version has two new functions name and getScriptEngineClassLoader, these new functions have been implemented in the SimplePlatformProvider class.

File: oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/Resteasy3Provider.java
Comment (anuragkumawat) [reply oauth-resource-server/authorization-server/src/main/java/com/baeldung/authserver/config/Resteasy3Provider.java:1]: ResteasyProviderFactory class in new keycloak version no longer support clearContextData and pushContext methods, instead we need to use ResteasyContext class, also Dispatcher class has been moved from core to spi package.

File: oauth-resource-server/authorization-server/pom.xml (hunk 1 above)
Comment (anuragkumawat) [reply oauth-resource-server/authorization-server/pom.xml:132]: Unneeded code has been removed

**Pull request merged**